orchestra/executor.py

```diff
@@ -348,8 +348,17 @@ def _try_group_orders(dependency_graph, group):
             for g1, g2 in zip(permutation, permutation[1:]):
                 # Add edge from all nodes in g1 to all nodes in g2
                 for a1, a2 in product(g1, g2):
-                    if a1 is not a2:
-                        depgraph_copy.add_edge(a1, a2)
+                    same_action = a1 is a2
+                    same_build = (
+                            isinstance(a1, ActionForBuild)
+                            and isinstance(a2, ActionForBuild)
+                            and a1.build is a2.build
+                    )
+                    # Don't add self loops or edges between actions for the same build
+                    if same_action or same_build:
+                        continue
```

Comment:
Can you explain why we want this?

Reply:
Yes, this handles an edge case where orchestra added a dependency edge between actions of the same build (e.g. causing a loop like `install A@1 -> configure A@1 -> install A@1`) which I found while developing the testsuite.
This condition guards against:
- adding edges that loopback to the same action
- adding edges between actions pertaining to the same build

We never want to add those kinds of edges, as in the best case they don't add any value, and in the worst case they add a cycle which we can't fix as there are no AnyOf nodes in it.
I'll expand the comment to clarify this.